ADVANCED_EXAMPLES.md

```diff
@@ -0,0 +1,594 @@
+```
+Create an instance, then clone it, and see the result:
+```js
+const logger = CloneablePrependlogger({ prefix: 'OUT: ' }); // creating first object
+const loggerClone = logger.clone(); // cloning the object.
+logger.log('hello'); // OUT: hello
+loggerClone.log('hello'); // OUT: hello
+```
+Prints
+```
+OUT: hello
+OUT: hello
+```
+Objects have the same state again. Awesome!
+
+### Attach function to prototype (memory efficient)
+
+Let's reimplement the `Cloneable` stamp so that the `clone()` function is not attached 
+to every object but to the prototype. This will save us a little bit of memory per object.
+```js
+const Cloneable = stampit.init(({ instance, stamp }) => {
+  if (!stamp.fixed.methods.clone) { // Avoid adding the same method to the prototype twice.
+    stamp.fixed.methods.clone = function () { return stamp(this); };
+  }
+});
+```
+The `stamp.fixed` property contains stamp's internal data.
+The `stamp.fixed.methods` object is used as all object instances' `.prototype`.
+Compose this new stamp with our `PrependLogger` from above:
+```js
+const CloneablePrependlogger = PrependLogger.compose(Cloneable);
+```
+Let's see how it works:
+```js
+const logger = CloneablePrependlogger({ prefix: 'OUT: ' });  // creating first object
+const loggerClone = logger.clone(); // cloning the object.
+logger.log('hello'); // OUT: hello
+loggerClone.log('hello'); // OUT: hello
+```
+Prints
+```
+OUT: hello
+OUT: hello
+```
+Memory efficient and safe cloning for each object. Yay!
+
+------------------------------
+
+## Delayed object instantiation using Promises
+
+> Run the examples below for yourself:
+```sh
+$ git clone https://github.com/stampit-org/stampit.git
+$ cd stampit && npm i babel -g
+$ babel-node advanced-examples/delayed-instantiation.js
+```
+
+What if you can't create an object right now but have to retrieve data from a server or filesystem?
+
+To solve this we can make **any** stamp to return `Promise` instead of object itself.
+
+First, let's assume you have this stamp:
+```js
+const User = stampit.refs({ entityName: 'user' });
+```
+
+The following stamp should be composed *last*, otherwise it won't work.
+```js
+const AsyncInitializable = stampit.refs({
+  db: { user: { getById() { return Promise.resolve({ name: { first: 'John', last: 'Snow' }}) } } } // mocking a DB
+}).methods({
+  getEntity(id) { // Gets id and return Promise which resolves into DB entity.
+    return Promise.resolve(this.db[this.entityName].getById(id));
+  }
+}).init(function () {
+  // If we return anything from an .init() function it becomes our object instance.
+  return this.getEntity(this.id);
+});
+```
+Let's compose it with our `User` stamp:
+```js
+const AsyncInitializableUser = User.compose(AsyncInitializable); // The stamp produces promises now.
+```
+Create object (ES6):
+```js
+const userEntity = AsyncInitializableUser({ id: '42' }).then(console.log);
+```
+A random stamp received the behaviour which creates objects asynchronously. OMG!
```

Comment:
:+1: 

[International standard = American English](http://www.openculture.com/2015/06/a-proportional-visualization-of-the-worlds-most-popular-languages.html)